x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/DenseVectorFieldTypeIT.java

```diff
@@ -42,32 +44,48 @@ public class DenseVectorFieldTypeIT extends AbstractEsqlIntegTestCase {
         "bbq_flat",
         "flat"
     );
+    public static final Set<String> NON_QUANTIZED_DENSE_VECTOR_INDEX_TYPES = Set.of(
+        "hnsw",
+        "flat"
+    );
 
+    private final ElementType elementType;
+    private final boolean synthetic;
     private final String indexType;
     private final boolean index;
-    private final boolean synthetic;
 
     @ParametersFactory
     public static Iterable<Object[]> parameters() throws Exception {
         List<Object[]> params = new ArrayList<>();
         // Indexed field types
-        for (String indexType : DENSE_VECTOR_INDEX_TYPES) {
-            params.add(new Object[] { indexType, true, false });
+        for (String indexType : ALL_DENSE_VECTOR_INDEX_TYPES) {
```

Comment:
Check all types for float and bytes. Quantized types only make sense for floats.